Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -60,73 +60,108 @@ ILogger<ITextDocumentLoader> logger
+      IReadOnlyDictionary<Uri, VerificationTree> migratedVerificationTrees,
+      CancellationToken cancellationToken) {
+#pragma warning disable CS1998
+      return await await DafnyMain.LargeStackFactory.StartNew(
+        async () => ResolveInternal(compilation, migratedVerificationTrees, cancellationToken), cancellationToken
+#pragma warning restore CS1998
         );
-      } catch (OperationCanceledException) {
-        return compilationAfterParsing;
+    }
+
+    private CompilationAfterResolution ResolveInternal(CompilationAfterParsing compilation,
+      IReadOnlyDictionary<Uri, VerificationTree> migratedVerificationTrees, CancellationToken cancellationToken) {
+
+      var program = compilation.Program;
+      var errorReporter = (DiagnosticErrorReporter)program.Reporter;
+      if (errorReporter.HasErrors) {
+        throw new TaskCanceledException();
+      }
+
+      var project = compilation.Project;
+
+      _ = statusPublisher.SendStatusNotification(compilation, CompilationStatus.ResolutionStarted);
+      var compilationUnit = symbolResolver.ResolveSymbols(project, program, cancellationToken);
+      var legacySymbolTable = symbolTableFactory.CreateFrom(compilationUnit, cancellationToken);
+
+      var newSymbolTable = errorReporter.HasErrors
+        ? null
+        : symbolTableFactory.CreateFrom(program, compilation, cancellationToken);
+      if (errorReporter.HasErrors) {
+        _ = statusPublisher.SendStatusNotification(compilation, CompilationStatus.ResolutionFailed);
+      } else {
+        _ = statusPublisher.SendStatusNotification(compilation, CompilationStatus.CompilationSucceeded);
       }
+
+      var ghostDiagnostics = ghostStateDiagnosticCollector.GetGhostStateDiagnostics(legacySymbolTable, cancellationToken);
+
+      List<ICanVerify> verifiables;
+      if (errorReporter.HasErrorsUntilResolver) {
+        verifiables = new();
+      } else {
+        var symbols = SymbolExtensions.GetSymbolDescendants(program.DefaultModule);
+        verifiables = symbols.OfType<ICanVerify>().Where(v => !AutoGeneratedToken.Is(v.RangeToken) &&
+                                                              v.ContainingModule.ShouldVerify(program.Compilation) &&
+                                                              v.ShouldVerify(program.Compilation) &&
+                                                              v.ShouldVerify).ToList();
+      }
+
+      return new CompilationAfterResolution(compilation,
+        errorReporter.AllDiagnosticsCopy,
+        newSymbolTable,
+        legacySymbolTable,
+        ghostDiagnostics,
+        verifiables,
+        new(),
+        new()
+      );
     }
 
     private IdeState CreateDocumentWithEmptySymbolTable(Compilation compilation,
       IReadOnlyDictionary<Uri, IReadOnlyList<Diagnostic>> resolutionDiagnostics) {
       var dafnyOptions = DafnyOptions.Default;
       var program = new EmptyNode();
       return new IdeState(
+        compilation.Version,
```

Comment:
Isn't that what was sufficient to ensure that gutter icons are handled in order?

Reply:
Can you elaborate?

Reply:
It looks like before the Ide state did not have this field and I remember you mentioning to me that because versioning had been lost, gutter icons were sometimes published in the wrong order. I was wondering if this fixes that.